### PR TITLE
fix(computed-fks): using computed foreign keys requires implementing the Present operator

### PR DESCRIPTION
**Pull request by romain-gilliotte**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/agent/test/routes/modification/delete-related.test.ts [hunk 1]
```diff
@@ -428,6 +427,7 @@ describe('DissociateDeleteRelatedRoute > delete', () => {
                   ],
                   field: 'myLibrary:id',
                 }),
+                factories.conditionTreeLeaf.build({ operator: 'Present', field: 'libraryId' }),
```
Comment (Scra3) [inline packages/agent/test/routes/modification/delete-related.test.ts:430]: Why lines are moved ?

File: docs/agent-customization/relationships/computed-fks.md [hunk 2]
```diff
@@ -33,7 +34,7 @@ agent.customizeCollection('customers', collection => {
     },
   });
 
-  // Implement the 'In' operator (required).
+  // Implement the 'In' operator.
```
Comment (Scra3) [inline docs/agent-customization/relationships/computed-fks.md:37]: why removed the '(required)' if it is the case ?

File: packages/datasource-toolkit/src/interfaces/query/filter/factory.ts [hunk 3]
```diff
@@ -183,10 +189,21 @@ export default class FilterFactory {
...
-        new ConditionTreeLeaf(relation.foreignKey, 'Present'),
+      const foreignKeySchema = through.schema.fields[relation.foreignKey];
+      let throughTree: ConditionTree = new ConditionTreeLeaf(
+        relation.originKey,
+        'Equal',
+        originValue,
       );
+
+      // Handle null foreign key case only when the datasource supports it.
+      if (foreignKeySchema.type === 'Column' && foreignKeySchema.filterOperators.has('Present')) {
```
Comment (Scra3) [inline packages/datasource-toolkit/src/interfaces/query/filter/factory.ts:200]: Instead to use 'Present', can we replace It by the 'NotEqual' operator ? If it is possible, we can apply the condition for all the cases. And removed the `filter` that removes the 'null' values.
`new ConditionTreeLeaf(relation.foreignKey, 'NotEqual', null),`

Comment (romain-gilliotte) [reply packages/datasource-toolkit/src/interfaces/query/filter/factory.ts:200]: If `Present` is not supported, we know that `NotEqual` is not supported either.
They always go together (thanks to the `OperatorReplacement` operator)

File: docs/agent-customization/relationships/computed-fks.md (hunk 2 above)
Comment (romain-gilliotte) [reply docs/agent-customization/relationships/computed-fks.md:37]: I added text on top to specify that all three steps are required
It seemed redundant

File: packages/agent/test/routes/modification/delete-related.test.ts (hunk 1 above)
Comment (romain-gilliotte) [reply packages/agent/test/routes/modification/delete-related.test.ts:430]: because the code changed :)
look at the CollectionUrl in the toolkit

**Review by Scra3 — APPROVED** [review]

**Pull request merged**